aten/src/ATen/native/BinaryOps.cpp

```diff
@@ -58,13 +58,27 @@ Tensor& add_(Tensor& self, const Tensor& other, Scalar alpha) {
 }
 
 Tensor& div_out(Tensor& result, const Tensor& self, const Tensor& other) {
+  if (isIntegralType(result.scalar_type(), /*includeBool=*/ true)) {
+    TORCH_WARN_ONCE(
+      "Integer division of tensors using div or / is deprecated, ",
+      "and in a future release div will perform true division as in Python 3. ",
+      "Use true_divide or floor_divide (// in Python) instead.");
```

Comment:
should we say something about "import __future__"?

Reply:
It would leave master in a bad state. I'll rebase https://github.com/pytorch/pytorch/pull/34794 once this is in and update the doc.